ast-visual-studio-extension/CxExtension/Services/ASCAUIManager.cs

```diff
@@ -0,0 +1,312 @@
+using Microsoft.VisualStudio.TextManager.Interop;
+using MARKERTYPE = Microsoft.VisualStudio.TextManager.Interop.MARKERTYPE;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell.Interop;
+using ast_visual_studio_extension.CxExtension.Utils;
+
+
+namespace ast_visual_studio_extension.CxExtension.Services
+{
+    public class ASCAUIManager
+    {
+        private readonly DTE2 _dte;
+        private OutputWindowPane _outputPane;
+        private ErrorListProvider _errorListProvider;
+        private List<IVsTextLineMarker> _activeMarkers = new List<IVsTextLineMarker>();
+
+        public ASCAUIManager()
+        {
+            _dte = Package.GetGlobalService(typeof(DTE)) as DTE2;
+            InitializeOutputPane();
+        }
+
+        public DTE2 GetDTE()
+        {
+            return _dte;
+        }
+
+        private void InitializeOutputPane()
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            var outputWindow = _dte.ToolWindows.OutputWindow;
+            _outputPane = OutputPaneUtils.InitializeOutputPane(outputWindow, CxConstants.EXTENSION_TITLE);
+        }
+
+        public Document GetActiveDocument()
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            return _dte?.ActiveDocument;
+        }
+
+        public void WriteToOutputPane(string message)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            _outputPane?.OutputString($"{DateTime.Now}: {message}\n");
+        }
+
+        public async Task DisplayDiagnosticsAsync(List<CxAscaDetail> scanDetails, string filePath)
+        {
+            if (scanDetails == null || string.IsNullOrEmpty(filePath)) return;
+
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            var document = _dte.Documents.Cast<Document>()
+                .FirstOrDefault(doc => doc.FullName == filePath);
+
+            if (document == null) return;
+
+            if (_errorListProvider == null)
+            {
+                _errorListProvider = new ErrorListProvider(ServiceProvider.GlobalProvider);
+            }
+
+            _errorListProvider.Tasks.Clear();
+            ClearAllMarkers();
+
+            var textManager = ServiceProvider.GlobalProvider.GetService(typeof(SVsTextManager)) as IVsTextManager2;
+            if (textManager == null) return;
+
+            IVsTextView activeTextView = null;
+            IVsTextLines buffer = null;
+
+            try
+            {
+                // Get primary view (1), no buffer filter (null), reserved value (0)
+                int hr = textManager.GetActiveView2(1, null, 0, out activeTextView);
+                if (ErrorHandler.Failed(hr) || activeTextView == null) return;
+
+                hr = activeTextView.GetBuffer(out buffer);
+                if (ErrorHandler.Failed(hr) || buffer == null) return;
+
+                WriteToOutputPane($"{scanDetails.Count} security best practice violations were found in {document.FullName}");
```

Comment:
if scanDetails.Count == 1 should be "security best practice violation"

Reply:
Done